source/run-command/run-command.rst

```diff
@@ -187,7 +193,157 @@ Drivers MUST document the behavior of RunCommand if a ``maxTimeMS`` field  is al
 
+
+--------------------
+``runCursorCommand``
+--------------------
+
+The following represents how a runCursorCommand API may be exposed.
+
+.. code:: typescript
+
+    interface Db {
+      /**
+       * Takes an argument representing an arbitrary BSON document and executes it against the server.
+       */
+      runCursorCommand(command: BSONDocument, options: RunCursorCommandOptions): RunCommandCursor;
+    }
+
+    interface RunCursorCommandOptions extends RunCommandOptions {
+      /**
+       * For operations that create cursors, timeoutMS can either cap the lifetime of the cursor or be applied separately to the original operation and all next calls.
+       * To support both of these use cases, these operations MUST support a timeoutMode option.
+       *
+       * @defaultValue 'CURSOR_LIFETIME'
+       *
+       * @see https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.rst
+       */
+      timeoutMode?: 'ITERATION' | 'CURSOR_LIFETIME';
+
+      /**
+       * Identifies the type of cursor this is.
+       *
+       * A tailable cursor can receive empty ``nextBatch`` arrays in getMore responses.
+       * However, subsequent ``getMore`` operations may return documents if new data has become available.
+       *
+       * A tailableAwait cursor is an enhancement where instead of dealing with empty responses the server will block until data become available.
+       *
+       * @defaultValue NON_TAILABLE
+       */
+      cursorType: NON_TAILABLE | TAILABLE | TAILABLE_AWAIT;
+    }
+
+RunCursorCommand implementation details
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+RunCursorCommand provides a way to access MongoDB server commands that return a cursor directly without requiring a driver to implement a bespoke cursor implementation.
+The API is intended to be built upon RunCommand and take a document from a user and apply a number of common driver internal concerns before forwarding the command to a server.
+A driver can expect that the result from running this command will return a document with a ``cursor`` field and MUST provide the caller with a language native abstraction to continue iterating the results from the server.
+
+High level RunCursorCommand steps:
+
+* If no session is provided by the caller, create a ClientSession
+* Create a local Cursor instance
+* Run the cursor creating command provided by the caller
+* Store the ``firstBatch``, ``ns``, and ``id`` from the server's response as well as which server the operation was executed on
+* When the current batch has been fully iterated, using the same server the initial command was executed on execute a ``getMore``
+* Store the results of the getMore's ``nextBatch`` and update the cursor's ``id``
+* Continue executing ``getMore`` commands until the cursor is exhausted
+
+Driver Sessions
```

Comment:
If you'd like to add a reference to this section (like you have elsewhere), you can link to [Sessions and Cursors](https://github.com/mongodb/specifications/blob/master/source/sessions/driver-sessions.rst#sessions-and-cursors).